test/controllers/helpers/SupervisingCustomsOfficeHelperSpec.scala

```diff
@@ -60,4 +70,51 @@ class SupervisingCustomsOfficeHelperSpec extends UnitSpec with ExportsDeclaratio
       }
     }
   }
+
+  "SupervisingCustomsOfficeHelper on nextPage" should {
+    onJourney(DeclarationType.STANDARD, DeclarationType.SUPPLEMENTARY) { request =>
+      "goto InlandTransportDetailsController for STANDARD & SUPPLEMENTARY journeys" in {
+        nextPage(request)(Normal) mustBe routes.InlandTransportDetailsController.displayPage(Normal)
+      }
+    }
+
+    onJourney(DeclarationType.SIMPLIFIED, DeclarationType.OCCASIONAL) { request =>
+      "goto ExpressConsignmentController for SIMPLIFIED & OCCASIONAL journeys" in {
+        nextPage(request)(Normal) mustBe routes.ExpressConsignmentController.displayPage(Normal)
+      }
+    }
+
+    "for CLEARANCE journeys" when {
+      onJourney(DeclarationType.CLEARANCE) { request =>
+        skipDepartureTransportPageCodes.foreach { modeOfTransportCode =>
+          val cachedDec = aDeclaration(withType(request.declarationType), withBorderModeOfTransportCode(Some(modeOfTransportCode)))
+
+          s"transportLeavingBoarderCode is ${modeOfTransportCode}" should {
+            "goto ExpressConsignmentController" in {
+              nextPage(getRequest(cachedDec))(Normal) mustBe routes.ExpressConsignmentController.displayPage(Normal)
+            }
+          }
+        }
+
+        meaningfulModeOfTransportCodes
+          .filter(!skipDepartureTransportPageCodes.contains(_))
+          .foreach { modeOfTransportCode =>
+            val cachedDec = aDeclaration(withType(request.declarationType), withBorderModeOfTransportCode(Some(modeOfTransportCode)))
+
+            s"transportLeavingBoarderCode is ${modeOfTransportCode}" should {
```

Comment:
tbh, I do not like too much this way of nesting tests, as they end up to be not easily commented/debugged or run in isolation.

I would rather use a `And("...")` statement, if you want to notify `modeOfTransportCode`, and your loops inside the test.